src/control/server/storage/bdev/provider.go

```diff
@@ -177,30 +178,44 @@ func (resp *ScanResponse) filter(pciFilter ...string) *ScanResponse {
 // filtered by request "DeviceList" and empty filter implies allowing all.
 func (p *Provider) Scan(req ScanRequest) (resp *ScanResponse, err error) {
 	if p.shouldForward(req) {
+		var action string
 		req.DisableVMD = p.IsVMDDisabled()
 
 		p.Lock()
 		defer p.Unlock()
 		defer func() {
+			if err != nil {
+				return
+			}
+
+			msg := fmt.Sprintf("%s bdev scan cache (%d",
+				action, len(resp.Controllers))
+
 			if resp == nil {
+				p.log.Debugf("%s devices)", msg)
 				return
 			}
```

Comment:
This fmt.Sprintf() will segfault if resp is nil. Probably just need to put the test for nil before it.

Reply:
Thanks good catch (unlikely to occur in practice as we previously check for error but worth fixing anyway).

Taking the opportunity to merge in following PR https://github.com/daos-stack/daos/pull/3712 which moves inline `parse_ascii()` definition to `src/common/control.c` (where we can put any function definitions that need to be shared between control and data planes), this was requested by @NiuYawei . Hoping @jolivier23 can verify the `scons` changes look okay. 

#3712 also provides PCI address include list to SPDK on `scan --nvme-health` which avoids scan iteration for one instance trying to access devices claimed by the other running instance, fixes cosmetic issue which displays an "already claimed" error.